traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -248,7 +326,17 @@ sub delete {
 	if ( !defined($tenant) ) {
 		return $self->not_found();
 	}	
-	my $name = $self->db->resultset('Tenant')->search( { id => $id } )->get_column('name')->single();
+
+	my $parent_tenant = $tenant->parent_id;		
+	
+	my $tenant_utils = UI::TenantUtils->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db(undef);
+	
+	if (!$tenant_utils->is_tenant_resource_accessible($tenants_data, $parent_tenant)) {
```

Comment:
oh, that's a good point. if i try to delete my own tenant, i will get back "sorry, this tenant is assigned to a user", right?